From f98b59c7ef7a52fa69cd76bd1de5039cb0c98c47 Mon Sep 17 00:00:00 2001
From: Wangda Tan <wangda@apache.org>
Date: Fri, 1 May 2015 16:40:44 -0700
Subject: [PATCH 2528/2748] YARN-3006. Improve the error message when
 attempting manual failover with auto-failover
 enabled. (Akira AJISAKA via wangda)

(cherry picked from commit 7d46a806e71de6692cd230e64e7de18a8252019d)
(cherry picked from commit 4151e5729a1ed74ac94035d66876b2dcc52eaaaf)
(cherry picked from commit 3389bac42a17f74d939d24b414a4e1220cff5b9b)

Change-Id: If245e47d9ff1d3a6cc64898f48fbf475517d53ef
---
 .../main/java/org/apache/hadoop/ha/HAAdmin.java    |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/HAAdmin.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/HAAdmin.java
index bd6366c..96cf14d 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/HAAdmin.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/HAAdmin.java
@@ -285,7 +285,13 @@ private int failover(CommandLine cmd)
             "supported with auto-failover enabled.");
         return -1;
       }
-      return gracefulFailoverThroughZKFCs(toNode);
+      try {
+        return gracefulFailoverThroughZKFCs(toNode);
+      } catch (UnsupportedOperationException e){
+        errOut.println("Failover command is not supported with " +
+            "auto-failover enabled: " + e.getLocalizedMessage());
+        return -1;
+      }
     }
     
     FailoverController fc = new FailoverController(getConf(),
-- 
1.7.9.5

